include/itkImageToTorchImageFilter.hxx

```diff
@@ -18,48 +18,76 @@
 #ifndef itkImageToTorchImageFilter_hxx
 #define itkImageToTorchImageFilter_hxx
 
-#include "itkImageToTorchImageFilter.h"
+#include <torch/torch.h>
 
-#include "itkImageRegionIterator.h"
-#include "itkImageRegionConstIterator.h"
+#include "itkImageToTorchImageFilter.h"
 
 namespace itk
 {
 
-template <typename TInputImage, typename TOutputImage>
-ImageToTorchImageFilter<TInputImage, TOutputImage>
-::ImageToTorchImageFilter()
-{}
+template< typename TInputImage, typename TOutputTorchImage >
+void
+ImageToTorchImageFilter< TInputImage, TOutputTorchImage >
+::PrintSelf( std::ostream &os, Indent indent ) const
+{
+  Superclass::PrintSelf( os, indent );
+}
+
+
+template< typename TInputImage, typename TOutputTorchImage >
+void
+ImageToTorchImageFilter< TInputImage, TOutputTorchImage >
+::GenerateInputRequestedRegion()
+{
+  // Is this correct?!!!  Instead, is the superclass implementation sufficient?!!!
+  // Call the superclass's implementation of this method
+  Superclass::GenerateInputRequestedRegion();
+
+  // Get pointers to the input image.
+  typename Self::InputImageType *inputImage = const_cast< typename Self::InputImageType * >( this->GetInput( 0 ) );
+
+  if( inputImage != nullptr )
+    {
+    inputImage->SetRequestedRegionToLargestPossibleRegion();
+    }
+}
 
 
-template <typename TInputImage, typename TOutputImage>
+template< typename TInputImage, typename TOutputTorchImage >
 void
-ImageToTorchImageFilter<TInputImage, TOutputImage>
-::PrintSelf(std::ostream & os, Indent indent) const
+ImageToTorchImageFilter< TInputImage, TOutputTorchImage >
+::BeforeThreadedGenerateData()
 {
-  Superclass::PrintSelf(os, indent);
+  // Call the superclass' implementation of this method
+  Superclass::BeforeThreadedGenerateData();
+
+  typename Self::InputImageType const * const inputImage = this->GetInput();
+  const typename Self::InputImageSizeType inputSize = inputImage->GetBufferedRegion().GetSize();
+  typename Self::InputImagePixelType const * const inputBufferPointer = inputImage->GetBufferPointer();
+
+  typename Self::OutputTorchImageType * const outputTorchImage = this->GetOutput();
+  const typename Self::OutputTorchImageSizeType outputSize = outputTorchImage->GetBufferedRegion().GetSize();
+
+  itkAssertOrThrowMacro( inputSize == outputSize, "ImageToTorchImageFilter: input and output images must have the same size in each dimension");
```

Comment:
maybe add the word buffered region for clarification

Reply:
Will do.